address_lookup_tables/src/address_lookup_table_store.rs

```diff
@@ -0,0 +1,218 @@
+            .is_none()
+        {
+            LRPC_ALTS_IN_STORE.inc();
+        }
+        drop(lookup_table);
+    }
+
+    pub async fn reload_if_necessary(
+        &self,
+        alt_messages: &[&solana_sdk::message::v0::MessageAddressTableLookup],
+    ) {
+        let accounts_to_load = alt_messages
+            .iter()
+            .filter_map(|alt| match self.map.get(&alt.account_key) {
+                Some(alt_data) => {
+                    let size = alt_data.len() as u8;
+                    if alt.readonly_indexes.iter().any(|x| *x >= size)
+                        || alt.writable_indexes.iter().any(|x| *x >= size)
+                    {
+                        Some(alt.account_key)
+                    } else {
+                        None
+                    }
+                }
+                None => Some(alt.account_key),
+            })
+            .collect_vec();
+        self.load_alts_list(&accounts_to_load).await;
+    }
+
+    pub async fn reload_alt_account(&self, address: &Pubkey) {
+        log::info!("Reloading {address:?}");
+
+        let account = match self
+            .rpc_client
+            .get_account_with_commitment(address, CommitmentConfig::processed())
+            .await
+        {
+            Ok(acc) => acc.value,
+            Err(e) => {
+                log::error!(
+                    "Error for fetching address lookup table {} error :{}",
+                    address.to_string(),
+                    e.to_string()
+                );
+                None
+            }
+        };
+        match account {
+            Some(account) => {
+                self.save_account(address, &account.data);
+            }
+            None => {
+                log::error!("Cannot find address lookup table {}", address.to_string());
+            }
+        }
+    }
+
+    async fn load_accounts(&self, alt: &Pubkey, accounts: &[u8]) -> Option<Vec<Pubkey>> {
+        let alt_account = self.map.get(alt);
+        match alt_account {
+            Some(alt_account) => Some(
+                accounts
+                    .iter()
+                    .map(|i| alt_account[*i as usize])
+                    .collect_vec(),
+            ),
+            None => {
+                log::error!("address lookup table {} was not found", alt);
+                None
+            }
+        }
+    }
+
+    pub async fn get_accounts(&self, alt: &Pubkey, accounts: &[u8]) -> Vec<Pubkey> {
+        match self.load_accounts(alt, accounts).await {
+            Some(x) => x,
+            None => {
+                // forget alt for now, start loading it for next blocks
+                // loading should be on its way
+                vec![]
+            }
+        }
+    }
+
+    pub fn serialize_binary(&self) -> Vec<u8> {
+        bincode::serialize::<BinaryALTData>(&BinaryALTData::new(&self.map)).unwrap()
+    }
+
+    pub fn load_binary(&self, binary_data: Vec<u8>) {
```

Comment:
maybe rename to _load_from_binary_.
pls clarify if the method is designed to get called only once or multiple times